File: bootstrap-ui-tests/src/test/ruby/pages/login_page.rb
class LoginPage < SitePrism::Page
    url = ENV['BOOTSTRAP_URL'] || 'localhost:8080'

	set_url "http://#{ url }/#/login"
	
	element :login_field, "#login"
	element :password_field, "#password"
	element :login_button, "button[type=submit]"
	element :login_link, "a[text()='Login']"
	
	def open()
		login_link.click
	end
	
	def login(user, pass)
		login_field.set user
		password_field.set pass
		login_button.click
	end
	
	def loginTestUser()
		login("kinga", "12qwQW!@")
	end
end

File: bootstrap-ui-tests/src/test/scala/uitest/pages/LoginPage.scala
package uitest.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import uitest.commands.SeleniumCommands;

class LoginPage(driver: WebDriver) {
  val url = "http://localhost:8080/#/login"

  private val sc: SeleniumCommands = new SeleniumCommands(driver)

  @FindBy(css = "#login")
  val loginField: WebElement = null

  @FindBy(css = "#password")
  val passwordField: WebElement = null

  @FindBy(css = "button[type=submit]")
  val loginButton: WebElement = null

  def login(login: String, password: String){
      loginField.sendKeys(login);
      passwordField.sendKeys(password);
      loginButton.click();
      sc.waitForFinishLoading();
  }

  def openLoginPage(){
      driver.get(url)
      sc.waitForFinishLoading();
  }
}